psu/psu.c

```diff
@@ -227,6 +236,103 @@ static int sdp_errStatus(hid_device *dev)
 }
 
 
+typedef struct {
+	unsigned char reportID;
+	unsigned char padding;
+	unsigned short size;
+	unsigned char payload[];
+} __attribute__((packed)) mcuboot_frame_t;
+
+
+typedef struct {
+	unsigned char tag;
+	unsigned char flags;
+	unsigned char reserved;
+	unsigned char paramcnt;
+	int params[];
+} __attribute__((packed)) mcuboot_cmd_t;
+
+
+static unsigned short size2LE(unsigned short size)
+{
+	if (size == htons(size)) {
+		return (unsigned short)((size & (unsigned short)0xff) << 8) | (size >> 8);
+	}
+
+	return size;
+}
+
+
+static int mcuboot_getProperty(hid_device *dev, int which)
+{
+	unsigned char b[sizeof(mcuboot_frame_t) + sizeof(mcuboot_cmd_t) + 2 * 4] = { 0 };
+	mcuboot_frame_t *frame = (mcuboot_frame_t *)b;
+	mcuboot_cmd_t *cmd = (mcuboot_cmd_t *)(b + sizeof(mcuboot_frame_t));
+	int rc;
+
+	frame->reportID = FRAME_CMD_OUT;
+	frame->size = size2LE(sizeof(b) - sizeof(mcuboot_frame_t));
+
+	cmd->tag = MCU_GET_PROPERTY;
+	cmd->paramcnt = 2;
```

Comment:
In the documentation there is description that:
_The parameter count field in the header is always set to 2, for status code and command tag parameters_
Maybe this magic number should be a constant define ?

Reply:
I think it's ok to leave it as number, we define only 2 parameters next it anyway

Reply:
Ok